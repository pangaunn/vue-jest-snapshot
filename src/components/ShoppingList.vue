<template>
  <ul class="shopping-list">
    <li v-for="(item, index) in items" :key="item.uuid" :class="{ done: item.done}">
      {{ item.text }}
      <button @click="complete(index)">Done</button>
    </li>
    <input type="text" v-model="newItem" id="newItem"/><button @click="add" id="addBtn">+</button>
  </ul>
</template>

<script>
export default {
  name: 'shopping-list',
  data () {
    return {
      newItem: '',
      items: [
        {text: 'Beer', done: true, uuid: 1},
        {text: 'Milk', done: false, uuid: 2},
        {text: 'Apple', done: false, uuid: 3}
      ]
    }
  },
  methods: {
    complete (index) {
      this.items[index].done = true
    },
    add () {
      if (this.newItem !== '') {
        this.items.push({
          text: this.newItem,
          done: false
        })
      }
    }
  }
}
</script>
