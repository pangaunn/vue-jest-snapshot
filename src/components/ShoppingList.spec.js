import Vue from 'vue/dist/vue.common'
import ShoppingList from '../../src/components/ShoppingList.vue'

describe('ShoppingList component', () => {
  let $mounted
  beforeEach(() => {
    $mounted = new Vue(ShoppingList).$mount()
  })

  test('snapshot', () => {
    expect(true).toMatchSnapshot(true)
    let $html = $mounted.$el.outerHTML
    expect($html).toMatchSnapshot()
    $mounted.items[1].done = true
    Vue.nextTick(() => {
      $html = $mounted.$el.outerHTML
      expect($html).toMatchSnapshot()

      $mounted.items[1].done = false
      Vue.nextTick(() => {
        $html = $mounted.$el.outerHTML
        expect($html).toMatchSnapshot()
      })
    })
  })

  test('done click', () => {
    let lis = $mounted.$el.querySelectorAll('li')
    let button = lis[2].querySelector('button')
    let customEvent = new Event('click')
    button.dispatchEvent(customEvent)
    expect($mounted.items[2].done).toBeTruthy()
    Vue.nextTick(() => {
      let $html = $mounted.$el.outerHTML
      expect($html).toMatchSnapshot()
    })
  })

  test('add new item', () => {
    $mounted.newItem = 'NEW ITEM'
    Vue.nextTick(() => {
      let button = $mounted.$el.querySelector('#addBtn').click()
      Vue.nextTick(() => {
        let $html = $mounted.$el.outerHTML
        expect($html).toMatchSnapshot()
      })
    })
  })
})
