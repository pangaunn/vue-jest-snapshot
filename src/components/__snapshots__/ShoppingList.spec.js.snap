// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList component add new item 1`] = `
"<ul class=\\"shopping-list\\"><li class=\\"done\\">
      Beer
      <button>Done</button></li><li class=\\"\\">
      Milk
      <button>Done</button></li><li class=\\"\\">
      Apple
      <button>Done</button></li><li class=\\"\\">
      NEW ITEM
      <button>Done</button></li> <input type=\\"text\\" id=\\"newItem\\"><button id=\\"addBtn\\">+</button></ul>"
`;

exports[`ShoppingList component done click 1`] = `
"<ul class=\\"shopping-list\\"><li class=\\"done\\">
      Beer
      <button>Done</button></li><li class=\\"\\">
      Milk
      <button>Done</button></li><li class=\\"done\\">
      Apple
      <button>Done</button></li> <input type=\\"text\\" id=\\"newItem\\"><button id=\\"addBtn\\">+</button></ul>"
`;

exports[`ShoppingList component snapshot 1`] = `
"<ul class=\\"shopping-list\\"><li class=\\"done\\">
      Beer
      <button>Done</button></li><li class=\\"\\">
      Milk
      <button>Done</button></li><li class=\\"\\">
      Apple
      <button>Done</button></li> <input type=\\"text\\" id=\\"newItem\\"><button id=\\"addBtn\\">+</button></ul>"
`;

exports[`ShoppingList component snapshot 2`] = `
"<ul class=\\"shopping-list\\"><li class=\\"done\\">
      Beer
      <button>Done</button></li><li class=\\"done\\">
      Milk
      <button>Done</button></li><li class=\\"\\">
      Apple
      <button>Done</button></li> <input type=\\"text\\" id=\\"newItem\\"><button id=\\"addBtn\\">+</button></ul>"
`;

exports[`true 1`] = `true`;
